### PR TITLE
fix: Fix OOM error when a bad api key is passed

### PR DESCRIPTION
**Pull request by izaaz**
### Summary

When the server returns a 400 because of a bad api key, we continuously retry without dropping the data. This fixes that issue

Fixes issue #208

### Checklist

* [ ] Does your PR title have the correct [title format](https://github.com/amplitude/Amplitude-Kotlin/blob/main/CONTRIBUTING.md#pr-commit-title-conventions)?
* Does your PR have a breaking change?:

### CODE REVIEW

**Review by Mercy811 — APPROVED** [review]
Thank you Izaaz for fixing it. LGTM. Nits: add a test for in memory response handler

File: core/src/main/java/com/amplitude/core/utilities/FileResponseHandler.kt [hunk 1]
```diff
@@ -65,7 +65,7 @@ class FileResponseHandler(
             removeCallbackByInsertId(eventsString)
             throw e
         }
-        if (eventsList.size == 1) {
+        if (eventsList.size == 1 || badRequestResponse.isInvalidApiKeyResponse()) {
```
Comment (yuhao900914) [inline core/src/main/java/com/amplitude/core/utilities/FileResponseHandler.kt:68]: Here we remove the eventFile from the storage. 
Should we keep the event for an invalid API key instead?

Comment (izaaz) [reply core/src/main/java/com/amplitude/core/utilities/FileResponseHandler.kt:68]: You mean keep the event for a valid future API key?

Comment (yuhao900914) [reply core/src/main/java/com/amplitude/core/utilities/FileResponseHandler.kt:68]: Yes, keep it until the next try.  I saw [previous android SDK ](https://github.com/amplitude/Amplitude-Android/blob/main/src/main/java/com/amplitude/api/AmplitudeClient.java#L2337)just logged an error with invalid API.

Comment (crleona) [reply core/src/main/java/com/amplitude/core/utilities/FileResponseHandler.kt:68]: The issue is that we will continuously retry requests that are known to be bad, there's no case in which the instance can recover until restarted with a new API key, which will also effectively orphan these events as storage is also segmented by api key.

Comment (yuhao900914) [reply core/src/main/java/com/amplitude/core/utilities/FileResponseHandler.kt:68]: That makes sense. Thanks.

**Review by yuhao900914 — APPROVED** [review]

**Pull request merged**